### PR TITLE
Rewrite SSL Connection Logger

### PR DESCRIPTION
**Pull request by OferLevi85**
Signed-off-by: Ofer Levi <ofer.levi@verizonmedia.com>

### CODE REVIEW

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/log/jetty/JettyConnectionLogger.java [hunk 1]
```diff
@@ -79,83 +70,21 @@ protected void doStart() {
...
     @Override
     public void handshakeFailed(Event event, Throwable failure) {
         SSLEngine sslEngine = event.getSSLEngine();
         handleListenerInvocation("SslHandshakeListener", "handshakeFailed", "sslEngine=%h,failure=%s", Stream.of(sslEngine, failure).collect(Collectors.toList()), () -> {
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug(failure.getMessage());
-            }
-            ConnectionInfo info = sslToConnectionInfo.remove(sslEngine);
+            ConnectionInfo info = ConnectionInfo.from(sslEngine);
             if (info != null) {
```
Comment (havetisyan) [inline libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/log/jetty/JettyConnectionLogger.java:78]: when would info be null? looks like we're always going to get a valid info object so we should be able to remove this if check.

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/log/jetty/JettyConnectionLogger.java [hunk 2]
```diff
@@ -79,83 +70,21 @@ protected void doStart() {
...
                 info.setSslHandshakeFailure(failure);
             }
+
+            if (info.sslHandshakeFailureException != null) {
```
Comment (havetisyan) [inline libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/log/jetty/JettyConnectionLogger.java:82]: in this method I think we're always going to have the exception check so we should able to remove this check.

**Review by havetisyan — CHANGES_REQUESTED** [review]
A couple of other changes we should make as part of this PR:

a) JsonConnectionLogWriter - lines 41/42 - can we use try with resources for OutputStream so we don't need a finally block

b) JsonConnectionLogWriter - line 38 LOG is not used so let's remove it from the class

c) SSLConnectionLog - let's combine lines 34/35 - no need for a local variable

d) JettyConnectionLogger - where we have "this.sslHandshakeFailureMessage == GENERAL_SSL_ERROR" - it's probably safer to replace this with GENERAL_SSL_ERROR.equals(this.sslHandshakeFailureMessage)

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/log/jetty/JettyConnectionLogger.java (hunk 1 above)
Comment (OferLevi85) [reply libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/log/jetty/JettyConnectionLogger.java]: fixed

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/log/jetty/JettyConnectionLogger.java (hunk 2 above)
Comment (OferLevi85) [reply libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/log/jetty/JettyConnectionLogger.java]: fixed

**Comment by OferLevi85** [comment]
> A couple of other changes we should make as part of this PR:
> 
> a) JsonConnectionLogWriter - lines 41/42 - can we use try with resources for OutputStream so we don't need a finally block
> 
> b) JsonConnectionLogWriter - line 38 LOG is not used so let's remove it from the class
> 
> c) SSLConnectionLog - let's combine lines 34/35 - no need for a local variable
> 
> d) JettyConnectionLogger - where we have "this.sslHandshakeFailureMessage == GENERAL_SSL_ERROR" - it's probably safer to replace this with GENERAL_SSL_ERROR.equals(this.sslHandshakeFailureMessage)

fixed

**Review by havetisyan — APPROVED** [review]

**Pull request merged**